src/js/themes/base.d.ts

```diff
@@ -12,8 +18,13 @@ import {
   OpacityType,
   RoundType,
   PadType,
+  PropsOf,
 } from '../utils';
-import { TextProps } from '../components/Text';
+
+import { Anchor } from '../components/Anchor';
+import { Button } from '../components/Button';
```

Comment:
Nice catch! I've removed the import.

RE: ExtendType type param

Originally I was thinking about adding the type param everywhere but after some investigation I found that in many components a different set of props are passed to the `styled` component. For example, in [Button](https://github.com/grommet/grommet/blob/1cf786ddafe7bf3329f9a37bb0bc51ebac4b2ec7/src/js/components/Button/Button.js#L280) it looks like a prop called `sizeProp` is passed instead of `size`, etc.

I wasn't sure yet how to handle that, so I changed my mind about adding all the types until it's clearer how we can tell which props will be present when the extend function is called, that way the types aren't saying that props will be present that won't really be there.

To simplify things I thought about just removing the type param altogether, which would leave the props typed as `ThemeStyledProps<any, ThemeType>`, and leaving it up to callers to determine what props are available, but it seemed useful to progressively add more typing to these in the future.

What do you think?

Reply:
I was torn on this whole bit too, went back and forth but everything i could think of came with additional questions. I think its good for now!